### PR TITLE
"local embeddings"

### PR DESCRIPTION
**Pull request by lleo-13**
Changed the structure of the embeddings service to a ROS2 structure and removed the client service for a local one.

### CODE REVIEW

File: ws/src/frida_language_processing/scripts/item_categorization.py [hunk 1]
```diff
@@ -1,152 +1,98 @@
 #!/usr/bin/env python3
 """
-ROS node for the Storing groceries task, should return the category of an item or list of items
+ROS2 node that returns the best match of an item or list of items
 """
-import rospy
+import rclpy
 import pandas as pd
```
Comment (Oscar-gg) [inline ws/src/frida_language_processing/scripts/item_categorization.py]: Remove unused dependencies

File: ws/src/frida_language_processing/scripts/item_categorization.py [hunk 2]
```diff
@@ -1,152 +1,98 @@
...
-import rospkg
-from openai import OpenAI
-
-from frida_hri_interfaces.srv import ItemsCategory, ItemsCategoryResponse
-
+import time
+import pynvml
+from frida_hri_interfaces.srv import ItemsCategory
+from ament_index_python.packages import get_package_share_directory
+import pynvml
```
Comment (Oscar-gg) [inline ws/src/frida_language_processing/scripts/item_categorization.py]: Duplicated import

File: ws/src/frida_language_processing/scripts/item_categorization.py [hunk 3]
```diff
@@ -1,152 +1,98 @@
...
+        """Service to return the best match of the items given"""
+        start_time = time.time()
+        response = ItemsCategory.Response()
+        i = 0
+        for item in req.items:
+            item_embedding = [self.get_local_embedding(item)]
+            possible_items = self.get_similar_items(item_embedding[0], self.items_dataframe)
+            response.matches[i] = possible_items.iloc[0]['name']
+            i = 0 + 1
+        response.gpu_usage = self.get_gpu_usage()
```
Comment (Oscar-gg) [inline ws/src/frida_language_processing/scripts/item_categorization.py]: Maybe we can avoid computing details of gpu usage and response.time as they add overhead

Comment (Oscar-gg) [reply ws/src/frida_language_processing/scripts/item_categorization.py]: Also, gpu metrics could be affected if it's not the only process using gpu

File: ws/src/frida_language_processing/scripts/item_categorization.py [hunk 4]
```diff
@@ -1,152 +1,98 @@
...
         with open(os.path.join(current_dir, "assets", "embeddings_items.pkl"), "rb") as f:
             self.items_dataframe = pickle.load(f)
 
         with open(os.path.join(current_dir, "assets", "embeddings_categories.pkl"), "rb") as f:
             self.category_embeddings = pickle.load(f)
-
-        rospy.spin()
+        
+        with open(os.path.join(current_dir, "assets", "embeddings_locations.pkl"), "rb") as f:
+            self.locations_embeddings = pickle.load(f)
```
Comment (Oscar-gg) [inline ws/src/frida_language_processing/scripts/item_categorization.py:43]: self.category_embeddings, self.locations_embeddings, and self.actions_embeddings seem to be unused throughout the code and I think their corresponding files are missing / not commited (embeddings_categories.pkl, embeddings_locations.pkl, embeddings_actions.pkl)

**Review by Oscar-gg — CHANGES_REQUESTED** [review]
Se ve bien, pero creo que se pueden cambiar unas cositas rápido y luego ya merge a main.

File: ws/src/frida_language_processing/scripts/item_categorization.py (hunk 1 above)
Comment (lleo-13) [reply ws/src/frida_language_processing/scripts/item_categorization.py]: pushed changes

File: ws/src/frida_language_processing/scripts/item_categorization.py (hunk 2 above)
Comment (lleo-13) [reply ws/src/frida_language_processing/scripts/item_categorization.py]: pushed changes

File: ws/src/frida_language_processing/scripts/item_categorization.py (hunk 3 above)
Comment (lleo-13) [reply ws/src/frida_language_processing/scripts/item_categorization.py]: pushed changes

File: ws/src/frida_language_processing/scripts/item_categorization.py (hunk 4 above)
Comment (lleo-13) [reply ws/src/frida_language_processing/scripts/item_categorization.py:43]: pushed changes

**Review by Oscar-gg — APPROVED** [review]
lgtm

**Pull request merged**